src/Files.App/Actions/Display/LayoutAction.cs

```diff
@@ -0,0 +1,179 @@
+
+		public override string Label { get; } = "MediumIcons".GetLocalizedResource();
+
+		public override RichGlyph Glyph { get; } = new("\uF0E2");
+		public override HotKey HotKey { get; } = new(VirtualKey.Number4, VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift);
+	}
+
+	internal class LayoutGridLargeAction : ToggleLayoutAction
+	{
+		protected override LayoutTypes LayoutType => LayoutTypes.GridLarge;
+
+		public override string Label { get; } = "LargeIcons".GetLocalizedResource();
+
+		public override RichGlyph Glyph { get; } = new("\uE739");
+		public override HotKey HotKey { get; } = new(VirtualKey.Number5, VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift);
+	}
+
+	internal class LayoutColumnsAction : ToggleLayoutAction
+	{
+		protected override LayoutTypes LayoutType => LayoutTypes.Columns;
+
+		public override string Label { get; } = "Columns".GetLocalizedResource();
+
+		public override RichGlyph Glyph { get; } = new(opacityStyle: "ColorIconColumnsLayout");
+		public override HotKey HotKey { get; } = new(VirtualKey.Number6, VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift);
+	}
+
+	internal class LayoutAdaptiveAction : ToggleLayoutAction
+	{
+		protected override LayoutTypes LayoutType => LayoutTypes.Adaptive;
+
+		public override string Label { get; } = "Adaptive".GetLocalizedResource();
+
+		public override bool IsExecutable => Context.IsLayoutAdaptiveEnabled;
+
+		public override RichGlyph Glyph { get; } = new("\uF576");
+		public override HotKey HotKey { get; } = new(VirtualKey.Number7, VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift);
+
+		protected override void OnContextChanged(string propertyName)
+		{
+			if (propertyName is nameof(IDisplayPageContext.IsLayoutAdaptiveEnabled))
+				OnPropertyChanged(nameof(IsExecutable));
+		}
+	}
+
+	internal abstract class ToggleLayoutAction : ObservableObject, IToggleAction
+	{
+		protected IDisplayPageContext Context { get; } = Ioc.Default.GetRequiredService<IDisplayPageContext>();
+
+		protected abstract LayoutTypes LayoutType { get; }
+
+		public abstract string Label { get; }
+
+		public abstract RichGlyph Glyph { get; }
+		public abstract HotKey HotKey { get; }
+
+		private bool isOn;
+		public bool IsOn => isOn;
+
+		public virtual bool IsExecutable => true;
+
+		public ToggleLayoutAction()
+		{
+			isOn = Context.LayoutType == LayoutType;
+			Context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		public Task ExecuteAsync()
+		{
+			Context.LayoutType = LayoutType;
+			return Task.CompletedTask;
+		}
+
+		private void Context_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName is nameof(IDisplayPageContext.LayoutType))
+				SetProperty(ref isOn, Context.LayoutType == LayoutType, nameof(IsOn));
+
+			if (e.PropertyName is not null)
+				OnContextChanged(e.PropertyName);
+		}
+
+		protected virtual void OnContextChanged(string propertyName) { }
+	}
+
+	internal class LayoutPreviousAction : IAction
```

Comment:
What does this do?

Reply:
It is a command that allows you to navigate between all the layouts. This is not used but may be available in the future command handler or with a new or custom hotkey.

Reply:
We already have `ctrl` + `+` and `ctrl` + `-`.

Reply:
I replaced Previous/Next with DecreaseSize/IncreaseSize.